node.go

```diff
@@ -612,6 +611,9 @@ func (node *Node) writeBytes(w io.Writer) error {
 		if node.leftNodeKey == nil {
 			return ErrLeftNodeKeyEmpty
 		}
+		if node.rightNodeKey == nil {
```

Comment:
was this a pre-existing bug or something required for the new logic?

Reply:
This was a guard that was left out when the v0 logic was ported to v1